### PR TITLE
REPL doc lookup assumed ASCII for the given string

### LINKED ISSUE #41589: Help/documentation for ÷= is broken
When running `?÷=` in the REPL, the lookup fails with this error:
```julia
help?> ÷=
ERROR: LoadError: StringIndexError: invalid index [2], valid nearby indices [1]=>'÷', [3]=>'='
Stacktrace:
 [1] string_index_err(s::String, i::Int64)
   @ Base ./strings/string.jl:12
 [2] getindex
   @ ./strings/string.jl:263 [inlined]
 [3] lookup_doc(ex::Symbol)
   @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/docview.jl:224
 [4] docm(source::LineNumberNode, mod::Module, ex::Any)
   @ Base.Docs ./docs/Docs.jl:521
 [5] var"@doc"(__source__::LineNumberNode, __module__::Module, x::Vararg{Any, N} where N)
   @ Core ./boot.jl:508
in expression starting at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/docview.jl:466
```
The error is a little confusing, since it doesn't seem that it can't *find* the documentation, but rather, it fails to index into the string "÷=".

Version info:
```
julia> versioninfo()
Julia Version 1.6.2
Commit 1b93d53fc4 (2021-07-14 15:36 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-9300H CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```

### PR DESCRIPTION
**Pull request by Seelengrab**
Fix #41589

Original code assumed ASCII, but that's not true. There may be other places in Base or an Stdlib where this happens, I'll grep and possibly open another PR if I find more.

Also, is there a better way to check whether something just doesn't throw? Checking the type of the returned value certainly seems like a way, but is of limited use for other code paths and kind of feels wrong...

### CODE REVIEW

**Comment by jakobnissen** [comment]
I'm pretty sure the indices `1:end-1` works for any string that ends in a `=`, so I'm not convinced the old behaviour is incorrect.

**Comment by louie-github** [comment]
A quick test shows this not to be the case:

```julia
julia> "÷="[end-1]
ERROR: StringIndexError: invalid index [2], valid nearby indices [1]=>'÷', [3]=>'='
Stacktrace:
 [1] string_index_err(s::String, i::Int64)
   @ Base .\strings\string.jl:12
 [2] getindex_continued(s::String, i::Int64, u::UInt32)
   @ Base .\strings\string.jl:233
 [3] getindex(s::String, i::Int64)
   @ Base .\strings\string.jl:226
 [4] top-level scope
   @ REPL[1]:1
```

File: stdlib/REPL/src/docview.jl [hunk 1]
```diff
@@ -221,7 +221,7 @@ function lookup_doc(ex)
         str = string(ex)
         isdotted = startswith(str, ".")
         if endswith(str, "=") && Base.operator_precedence(ex) == Base.prec_assignment && ex !== :(:=)
-            op = str[1:end-1]
+            op = str[begin:prevind(str, lastindex(str))]
```
Comment (KristofferC) [inline stdlib/REPL/src/docview.jl:224]: Is this equivalent to `chop(str)`?

Comment (fredrikekre) [reply stdlib/REPL/src/docview.jl:224]:
```suggestion
            op = chop(str)
```

File: stdlib/REPL/test/docview.jl [hunk 2]
```diff
@@ -47,3 +47,9 @@ end
     # shouldn't throw when there is a space in a middle of query
     @test (REPL.matchinds("a ", "a file.txt"); true)
 end
+
+@testset "Unicode doc lookup" begin
+    # https://github.com/JuliaLang/julia/issues/41589
+    # assuming ASCII is bad, m'kay?
```
Comment (fredrikekre) [inline stdlib/REPL/test/docview.jl:53]:
```suggestion
@testset "Unicode doc lookup (#41589)" begin
```

**Comment by Seelengrab** [comment]
> I'm pretty sure the indices 1:end-1 works for any string that ends in a =, so I'm not convinced the old behaviour is incorrect.

The byte index before a valid Char index may not be a valid Char index itself (except for single-byte encodings like ASCII). In this case, the character in front of `=` is a multi-byte character of length 2 bytes, that's why `end - 1` is not valid here.

---

Thanks for the quick review, did not know about `chop`!

**Comment by KristofferC** [comment]
> Should there be a comment next to the test, saying that the isa Markdwon.MD check is just used for confirming that it doesn't throw, instead of requiring that return type?

I don't think it really matters. Someone can always follow the issue -> PR trail from the testset name if they are really interested.

**Pull request merged**